### ISSUE TITLE
Download data based on query result

### ISSUE THREAD

**Issue by dvstans** [ORNL/DataFed#242]
Add ability to download data based on query results (direct or saved). Would need option to limit max/expected number of downloads.

**Comment by JoshuaSBrown** [comment]
I don't understand how this would work. This is so a user could for instance use a query on the metadata and then directly download the raw data that match the query? This is already available via the Python API right? So I'm guessing this is from the point of view of the Web UI?

**Comment by dvstans** [comment]
That is correct; however, it's not directly supported by the API. You have to harvest all the IDs from the query result then download them separately. This would be a convenience feature, but could be helpful for some use cases.
